opennms-container/common.mk

```diff
@@ -0,0 +1,206 @@
+	@echo "  install:   Load the OCI file in your Docker instance"
+	@echo "  uninstall: Remove the container image from your Docker instance"
+	@echo "  uninstall-all: Remove all container images built by this Makefile from your Docker instance"
+	@echo "  clean:     Remove the Dockerx build instance, keeping the files in the directory images/"
+	@echo "  clean-all: Remove the Dockerx build instance and delete *everything* in the directory images/"
+	@echo ""
+	@echo "Arguments to modify the build:"
+	@echo "  VERSION:            Version number for this release of the deploy-base artefact, default: $(VERSION)"
+	@echo "  BASE_IMAGE:         The base image we install our Java app as a tarball, default: $(BASE_IMAGE)"
+	@echo "  DOCKERX_INSTANCE:   Name of the docker buildx instance, default: $(DOCKERX_INSTANCE)"
+	@echo "  DOCKER_REGISTRY:    Registry to push the image to, default is set to $(DOCKER_REGISTRY)"
+	@echo "  DOCKER_ORG:         Organisation where the image should pushed in the registry, default is set to $(DOCKER_ORG)"
+	@echo "  DOCKER_PROJECT:     Name of the project in the registry, the default is set to $(DOCKER_PROJECT)"
+	@echo "  DOCKER_TAG:         Docker tag is generated from registry, org, project, version and build number, set to $(DOCKER_TAG)"
+	@echo "  DOCKER_ARCH:        Architecture for OCI image, default: $(DOCKER_ARCH)"
+	@echo "  DOCKER_OCI:         Path to OCI image, default: $(DOCKER_OCI)"
+	@echo "  DOCKER_OUTPUT:      Output method: $(DOCKER_OUTPUT)"
+	@echo "  DOCKER_OUTPUT_OCI:  DOCKER_OUTPUT value used to write a single architecture to a file, default: $(DOCKER_OUTPUT_OCI)"
+	@echo "  DOCKER_OUTPUT_IMAGE: DOCKER_OUTPUT value used to store the image in Docker: $(DOCKER_OUTPUT_IMAGE)"
+	@echo "  DOCKER_FLAGS:       Additional docker buildx flags, default: $(DOCKER_FLAGS)"
+	@echo "  BUILD_NUMBER:       In case we run in CI/CD this is the build number which produced the artifact, default: $(BUILD_NUMBER)"
+	@echo "  BUILD_URL:          In case we run in CI/CD this is the URL which for the build, default: $(BUILD_URL)"
+	@echo "  BUILD_BRANCH:       In case we run in CI/CD this is the branch of the build, default: $(BUILD_BRANCH)"
+	@echo ""
+	@echo "Example:"
+	@echo "  make build DOCKER_REGISTRY=myregistry.com DOCKER_ORG=myorg DOCKER_FLAGS=--push"
+	@echo ""
+
+# If we ever execute the commands in this recipe, it's because $(TARBALL)
+# doesn't exist and make wants to build it, so tell the user we couldn't
+# find the tarball and they need to run the assembly.
+$(TARBALL):
+	$(warning Couldn't find opennms-full-assembly tarball at $(TARBALL))
+	$(error Go to the top-level and run this: $(ASSEMBLE_COMMAND))
+
+test: $(TARBALL)
+	$(info Ready to go, let's light this candle!)
+	@true
+
+$(README): $(TARBALL)
+	@echo "Unpacking tarball for Docker context..."
+	mkdir -p tarball-root
+	tar -x -z --strip-components $(TAR_STRIP_COMPONENTS) -C ./tarball-root -f $<
+	touch $@
```

Comment:
Since this properly detects the tarball changing (at least, I think it should since you `touch` the README), shouldn't this `rm -rf tarball-root` before it unpacks? Otherwise it's possible a new tarball gets generated with different files in it, but the docker container ends up with a superset combining the two runs.

Reply:
Done!